### ISSUE TITLE
Error message when resources are exceeded is confusing

### ISSUE THREAD

**Issue by jcustenborder** [confluentinc/terraform-provider-confluentcloud#48]
The organization I was working with had 10 environments. When the provider went to great the 11th environment the following error message was returned. 

```
╷
│ Error: 402 Payment Required
│ 
│   with confluentcloud_environment.workshop,
│   on confluent.tf line 16, in resource "confluentcloud_environment" "workshop":
│   16: resource "confluentcloud_environment" "workshop" {
│ 
╵
```

**Comment by linouk23** [comment]
@jcustenborder could you confirm the version of TF Provider for Confluent Cloud that you're using? We did [add](https://github.com/confluentinc/terraform-provider-confluentcloud/blob/master/CHANGELOG.md#040-january-28-2022) descriptive error messages in `0.4.0` but there could be a bug or something.

**Comment by fireinhole** [comment]
I hit same 10 environment limit and getting the following message with version = "0.4.0"
```
│ Error: 402 Payment Required: Your organization is currently limited to 10 environments
│
│   with confluentcloud_environment.test-env,
│   on main.tf line 10, in resource "confluentcloud_environment" "test-env":
│   10: resource "confluentcloud_environment" "test-env" {
```

**Comment by linouk23** [comment]
@fireinhole it looks like there're 2 separate issues:
1. Not very descriptive error message: `402 Payment Required` -> `402 Payment Required: Your organization is currently limited to 10 environments`. (it seems to be resolved when updating from `0.3.0` to `0.4.0`).
2. How to resolve `Your organization is currently limited to 10 environments` [quotas](https://docs.confluent.io/cloud/current/quotas/index.html) issue which seems to be more like an API / quotas issue and not related to TF provider but let me check.
